pkgs/by-name/on/onscripter/package.nix

```
@@ -0,0 +1,80 @@
+{
+  lib,
+  SDL,
+  SDL_image,
+  SDL_mixer,
+  SDL_ttf,
+  bzip2,
+  fetchFromGitHub,
+  fontconfig,
+  freetype,
+  libjpeg,
+  libogg,
+  libpng,
+  libvorbis,
+  lua,
+  smpeg,
+  stdenv,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "onscripter";
+  version = "20230825";
+
+  src = fetchFromGitHub {
+    owner = "ogapee";
+    repo = "onscripter";
+    rev = finalAttrs.version;
+    hash = "sha256-XaZTtOkV+2dHmcgZ4GbyiMxRBYdwT5eHxx+r05eAmBw=";
+  };
+
+  patches = [
+    # Add "install" target to Makefile
+    ./add-make-install.patch
+    # 1. Remove libavifile dependency, which is old and unmaintained
+    # 2. Replace -llua5.1 with -llua because the former didn't work even with Lua 5.1
+    ./fix-linux-build.patch
+    # 1. Remove direct linker flags for macOS SDKs.
+    # 2. Build with Lua
+    #
+    # ONScripter appears to depend on macOS SDKs only indirectly through SDL, so flags provided by
+    # the sdl-config command should suffice. The existing flags also referenced the deprecated
+    # QuickTime framework, resulting in a build failure.
+    ./fix-darwin-build.patch
+  ];
+
+  nativeBuildInputs = [
+    freetype
+    SDL
+    smpeg
+  ];
+
+  buildInputs = [
+    SDL
+    SDL_image
+    SDL_mixer
+    SDL_ttf
+    bzip2
+    libjpeg
+    libogg
+    libpng
+    libvorbis
+    lua
+    smpeg
+  ]
+  ++ (if stdenv.isDarwin then [ freetype ] else [ fontconfig ]);
+
+  # The build script for BSDs also fall under Makefile.Linux
+  makefile = if stdenv.isDarwin then "Makefile.MacOSX" else "Makefile.Linux";
```

We are trying to deprecate `stdenv.isFoo` since it is ambiguous in the presence of cross‐compilation, so:

```suggestion
  ++ (if stdenv.hostPlatform.isDarwin then [ freetype ] else [ fontconfig ]);

  # The build script for BSDs also fall under Makefile.Linux
  makefile = if stdenv.hostPlatform.isDarwin then "Makefile.MacOSX" else "Makefile.Linux";
```